rdf_differ/domain/model.py

```diff
@@ -0,0 +1,86 @@
+"""
+model.py
+Date: 11/08/2020
+Author: Mihai Coșleț
+Email: coslet.mihai@gmail.com
+"""
+from dataclasses import dataclass
+from typing import Optional
+
+
+class VersionMissing(Exception):
+    pass
+
+
+class VersionExists(Exception):
+    pass
+
+
+class VersionsDeltaExists(Exception):
+    pass
+
+
+class RDFContentReference:
+    pass
+
+
+@dataclass
+class DatasetVersion:
+    version_id: str
+    description: Optional[str]
+    content_reference: Optional[RDFContentReference]
+
+    def __eq__(self, other: 'DatasetVersion'):
+        return self.version_id == other.version_id
+
+
+@dataclass
+class VersionsDelta:
+    old_version_id: str
+    new_version_id: str
+    insertions: Optional[RDFContentReference]
+    deletions: Optional[RDFContentReference]
+
+    def __eq__(self, other: 'VersionsDelta'):
+        return self.old_version_id == other.old_version_id and self.new_version_id == other.new_version_id
+
+
+class Dataset:
+    def __init__(self, name: str, uri: str, description: Optional[str] = ''):
+        self.name = name
+        self.uri = uri
+        self.description = description
+        self.versions = list()
+        self.version_deltas = list()
+
+    def add_version(self, dataset_version: DatasetVersion):
+        if self._version_exists(dataset_version.version_id):
+            raise VersionExists(f"This dataset version ({dataset_version.version_id}) already exists.")
+        self.versions.append(dataset_version)
+
+    def _version_exists(self, dataset_version: str) -> bool:
+        if dataset_version in [known_version.version_id for known_version in self.versions]:
+            return True
+        return False
+
+    def get_delta(self, old_version_id: str, new_version_id: str) -> Optional[VersionsDelta]:
+        try:
+            dummy_delta = VersionsDelta(old_version_id=old_version_id, new_version_id=new_version_id, insertions=None,
```

Comment:
Is `dummy_delta` a good name? I made me think that it's a fake object or some kind of test data tbh.